pkg/hive/manager.go

```diff
@@ -229,11 +229,12 @@ func (hr *clusterManager) ResetCorrelationData(ctx context.Context, doc *api.Ope
 			return err
 		}
 
-		err = utillog.ResetHiveCorrelationData(cd)
+		modified := cd.DeepCopy()
+		err = utillog.ResetHiveCorrelationData(modified)
 		if err != nil {
 			return err
 		}
 
-		return hr.hiveClientset.Update(ctx, cd)
+		return hr.hiveClientset.Patch(ctx, modified, client.MergeFrom(cd))
```

Comment:
I'm not sure. The specific field in question here (`"hive.openshift.io/additional-log-fields"` annotation) is a singular field that's expected to contain a stringified JSON map of all logging fields to add. It is possible for other clients to populate those fields, and the code here within the RP is structured to only modify the specific fields within that map that the RP manages, and it does not manage the entire map itself. 

Realistically, I don't think there's anything other than the RP in ARO's deployment of Hive that sets fields within that map, and observing real ClusterDeployment resources appears to back that up. 

IMO, we could set the RP to be the field manager on that annotation, but if we do that we should also update the code within the RP to actually take full ownership of that map and ensure it _only_ contains the log fields the RP expects to set, rather than just soft-enforcing its specific fields within the assumed-shared map.

Reply:
Thank you for the detailed thought, I agree if the RP is not managing the object, just ensuring we have specific annotations in the map, it does not seem like the RP needs to be the owner.

Reply:
I spent some time looking into this because I was having trouble understanding it, and I _think_ that even if we were to try to set `Force` here, it wouldn't work without some other changes since that field is meant for server-side apply, and we aren't using SSA here. See https://github.com/kubernetes-sigs/controller-runtime/blob/658c55227830b2d895b12fc1c86cbc731e36d291/pkg/client/options.go#L742 and https://github.com/kubernetes-sigs/controller-runtime/blob/658c55227830b2d895b12fc1c86cbc731e36d291/pkg/client/patch.go#L151

Reply:
Thanks for checking. Yeah I don't think this would work unless we also set up SSA on this call.